src/js/components/OptionalEnviromentComponent.jsx

```diff
@@ -0,0 +1,99 @@
+var React = require("react/addons");
+
+var appValidator = require("../validators/appValidator");
+var FormGroupComponent = require("../components/FormGroupComponent");
+var DuplicableRowControls = require("../components/DuplicableRowControls");
+
+var OptionalSettingsComponent = React.createClass({
+  displayName: "OptionalEnvironmentComponent",
+
+  propTypes: {
+    errors: React.PropTypes.array,
+    model: React.PropTypes.object.isRequired
+  },
+
+  getInitialState: function () {
+    var env = this.props.model.env;
+    var state = {
+      env: Object.keys(env).map(function (key) {
+        return {
+          key: key,
+          value: env[key]
+        };
+      })
+    };
+    if (state.env.length === 0) {
+      state.env = [true];
+    }
+    return state;
```

Comment:
So state.env can contain an array of {key,value}s, or true, or both when adding a row (see L#34).
A comment about what that represents would be really welcome.